test/fake_cloud_firestore_test.dart

```diff
@@ -135,6 +139,88 @@ void main() {
     }));
   });
 
+  group('Document Snapshots fires on deletion', () {
+    test('without converter', () async {
+      final instance = FakeFirebaseFirestore();
+      final collectionRef = instance.collection('users');
+      final docRef = collectionRef.doc(uid);
+      // Set document data. This does not fire an update since there is no
+      // listener yet.
+      await docRef.set({'name': 'Alice'});
+
+      // snapshots() fires once, some time after it returns a stream.
+      final allSnapshots = docRef.snapshots();
+
+      // We forcefully await the doc snapshot with data. Without this, there is a
+      // race condition where docRef.delete() runs before, and so snapshots()
+      // fires the first snapshot with no data.
+      expect(
+        allSnapshots,
+        emitsInOrder(
+          [
+            predicate<DocumentSnapshot>(
+              (d) => equals({'name': 'Alice'}).matches(
+                d.data(),
+                {},
+              ),
+            ),
+            predicate<DocumentSnapshot>((s) => !s.exists),
+          ],
+        ),
+      );
+
+      //Await the first snapshot to start the stream
+      await allSnapshots.first;
```

Comment:
I wonder if this workaround is still needed, since you introduced `_getSync` and `stream.startWith`. But you don't have to check. I will check and clean up later if needed.